### PR TITLE
[chore](audit) Optimize the SQL (insert into values) length in audit logs

### PR DESCRIPTION
**Pull request by yagagagaga**
## Proposed changes

- Add annotation information to the `insert into value` statement
- Let the SQL length in the audit log be controlled by `GlobalVariable.auditPluginMaxSqlLength`

Here is the result of this PR:
image

If the length of sql greater than `GlobalVariable.auditPluginMaxSqlLength`, the sql will be truncate but it still have a comment if the sql is `insert into value`

```sql
set global audit_plugin_max_sql_length=30;
insert into test.t values (1, 'Tom'), (2, 'Jelly'), (3, 'Damn');
```

image

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/qe/AuditLogHelper.java [hunk 1]
```diff
@@ -47,15 +57,52 @@ public class AuditLogHelper {
...
+            origStmt = handleStmt(origStmt, parsedStmt);
             logAuditLogImpl(ctx, origStmt, parsedStmt, statistics, printFuzzyVariables);
         } catch (Throwable t) {
             LOG.warn("Failed to write audit log.", t);
         }
     }
 
+    private static String handleStmt(String origStmt, StatementBase parsedStmt) {
+        int length = Math.min(GlobalVariable.auditPluginMaxSqlLength, origStmt.length());
+        origStmt = origStmt.replace("\n", " ").substring(0, length);
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/qe/AuditLogHelper.java:69]: \r\n

File: fe/fe-core/src/main/java/org/apache/doris/qe/AuditLogHelper.java [hunk 2]
```diff
@@ -47,15 +58,56 @@ public class AuditLogHelper {
...
         try {
+            origStmt = handleStmt(origStmt, parsedStmt);
             logAuditLogImpl(ctx, origStmt, parsedStmt, statistics, printFuzzyVariables);
         } catch (Throwable t) {
             LOG.warn("Failed to write audit log.", t);
         }
     }
 
+    public static String handleStmt(String origStmt, StatementBase parsedStmt) {
+        int length = Math.min(GlobalVariable.auditPluginMaxSqlLength, origStmt.length());
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/qe/AuditLogHelper.java:69]: return origstmt if origStmt.length < maxLength;

**Review by morrySnow — DISMISSED** [review]

File: regression-test/suites/audit/test_audit_log_behavior.groovy [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    for(int i = 0; i < cnt; i++) {
+        def tuple2 = sqls.get(i)
+        sql tuple2[0]
+    }
+
+    // check result
+    for(int i = 0; i < cnt; i++) {
+        def tuple2 = sqls.get(i)
+        def res = sql "select stmt from __internal_schema.audit_log where stmt like '%3F6B9A_${i}%' order by time asc limit 1"
+        while (res.isEmpty()) {
```
Comment (gavinchou) [inline regression-test/suites/audit/test_audit_log_behavior.groovy:79]: this will be a dead loop if something is wrong.

**Review by gavinchou — CHANGES_REQUESTED** [review]

**Comment by yagagagaga** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by LiBinfeng-01 — APPROVED** [review]

**Comment by yagagagaga** [comment]
run buildall

**Comment by yagagagaga** [comment]
run buildall

**Review by luwei16 — APPROVED** [review]

**Pull request merged**